pkg/pub_integration/test/search_update_test.dart

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) 2021, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:convert';
+
+import 'package:http/http.dart' as http;
+import 'package:pub_integration/src/fake_pub_server_process.dart';
+import 'package:pub_integration/src/headless_env.dart';
+import 'package:pub_integration/src/pub_puppeteer_helpers.dart';
+import 'package:puppeteer/puppeteer.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('browser', () {
+    late FakePubServerProcess fakePubServerProcess;
+    late final HeadlessEnv headlessEnv;
+    final httpClient = http.Client();
+
+    setUpAll(() async {
+      fakePubServerProcess = await FakePubServerProcess.start();
+      await fakePubServerProcess.started;
+    });
+
+    tearDownAll(() async {
+      await headlessEnv.close();
+      await fakePubServerProcess.kill();
+      httpClient.close();
+    });
+
+    test('bulk tests', () async {
+      final origin = 'http://localhost:${fakePubServerProcess.port}';
+      // init server data
+      await httpClient.post(Uri.parse('$origin/fake-test-profile'),
+          body: json.encode({
+            'testProfile': {
+              'defaultUser': 'admin@pub.dev',
+              'packages': List.generate(
+                100,
+                (i) => {
+                  'name': i % 3 == 0 ? 'flutter_pkg_$i' : 'pkg_$i',
+                  if (i % 11 == 0) 'publisher': 'example.com',
+                },
+              ),
+            },
+          }));
+      await httpClient.get(Uri.parse('$origin/fake-update-search'));
+
+      // start browser
+      headlessEnv = HeadlessEnv(testName: 'search-update', origin: origin);
+      await headlessEnv.startBrowser();
+
+      // listing page
+      await headlessEnv.withPage(
+        fn: (page) async {
+          await page.gotoOrigin('/experimental?enabled=1');
+          await page.gotoOrigin('/packages');
+
+          // check package list
+          final i1 = await listingPageInfo(page);
+          expect(i1.totalCount, 100);
+          expect(i1.packageNames, [
+            'pkg_11',
+            'flutter_pkg_51',
+            'pkg_13',
+            'pkg_62',
+            'pkg_34',
+            'pkg_28',
+            'pkg_94',
+            'flutter_pkg_66',
+            'flutter_pkg_0',
+            'pkg_73',
+          ]);
```

Comment:
Why is this the case?

Reply:
The test profile import uses a fake analysis by default, which assigns tags with a pseudorandom process (based on the hash of the package's name and sometimes version), with a few hardcoded patterns, e.g. `flutter_*` packages will get `sdk:flutter` tag assigned.
The test profile imported at the beginning creates 100 packages with semi-random tags, and adding and removing filters works because of the number of packages and their are kind of random.
(will add this as code comment)